spec/protocol/endpoints.adoc

```diff
@@ -1,19 +1,21 @@
 = MANY Endpoints
 
-A server exposes endpoints that a message must target.
-Endpoints receive a single bytestring payload, and optionally return either an error or a bytestring payload.
+Endpoints are methods exposed by servers that a client can call with a single bytestring payload.
+The server will reply to the client in a timely matter footnote:[If this cannot be done synchronously, the server MUST still respond with a response, see xref:../../attributes/network/8_async.adoc[Async Attribute].].
+The reply MAY contain the result of the operation or an error if the operation failed to execute properly.
 
 == Command Query Responsibility Segregation
 
 MANY separates messages into two kinds;
 
-1. Query (or read) which access data in read-only mode.
-   The app state hash should not be changed after running a query, and any queued command calls should be resolved or paused (commands SHOULD NOT be in the middle of executing).
+1. Query: access data in read-only mode.
+   Executing a query SHOULD NOT change the application state hash.
+   Any queued command calls should be resolved or paused (commands SHOULD NOT be in the middle of executing) prior to executing a query.
```

Comment:
```suggestion
   Any queued command calls should be resolved or paused before executing a query, i.e., the server SHOULD NOT execute a query while a command is running.
```

nit: I still don't like the `should be resolved or paused`

Reply:
Tried to reword removing that. We can wordsmith more later.